src/webgpu/api/validation/createBindGroupLayout.spec.ts

```diff
@@ -45,6 +45,32 @@ g.test('visibility_of_bindings_can_be_0').fn(async t => {
   });
 });
 
+g.test('storage_buffer_binding_support').fn(async t => {
```

Comment:
Let's try to parameterize this, something like the following:

```typescript
g.test('buffer_binding_valid_stages')
    params()
      .combine(poptions('type', kBufferBindingTypes))
      .combine(poptions('visibility', kShaderStageCombinations))
  )
  .fn(async t => {
    const { binding, visibility, type } = t.params;
    const typeInfo = kBufferBindingTypeInfo[type];

    const visibleOnlyToValidStages = (visibility & typeInfo.validStages) === visibility;

    const shouldError = !visibleOnlyToValidStages;
    t.expectValidationError(() => {
      t.device.createBindGroupLayout({
        entries: [{ binding: 0, visibility, type }],
      });
    }, shouldError);
  });
```

Reply:
Actually, while we're at it, we should make this test all binding types, not just buffer binding types, because nothing about this test is specific to buffer binding types.

That said, I just realized this has overlap with the test createPipelineLayout:visibility_and_dynamic_offsets, which, despite being in createPipelineLayout.spec.ts I'm pretty sure is actually testing createBindGroupLayout validation. We should just move it and tweak it slightly to look more like what I just wrote above, instead of using what I wrote above.

Reply:
Note that visibility_and_dynamic_offsets is actually written incorrectly right now (oops) - it wraps both createPipelineLayout and createBindGroupLayout in expectValidationError, so it's not correctly checking which of those two functions the error is in. One of the tweaks will be to fix that.

Reply:
@kainino0x , thanks for your review! You are right, visibility_and_dynamic_offsets actually tests createBindGroupLayout, not createPipelineLayout. And it can cover my previous code. So I just moved that test to createBindGroupLayout and removed my previous code. :). PTAL.